File: javascript/example_single_responsibility.js
class Report {
    constructor(title, content) {
        this.title = title;
        this.content = content;
    }

    generate() {
        return `Report: ${this.title}\n${this.content}`;
    }
}

const report = new Report('Monthly Sales', 'Sales increased by 20%');
console.log(report.generate());

File: java/ExampleSingleResponsibility.java
public class Report {
    private String title;
    private String content;

    public Report(String title, String content) {
        this.title = title;
        this.content = content;
    }

    public String generate() {
        return "Report: " + title + "\n" + content;
    }
}

public class Main {
    public static void main(String[] args) {
        Report report = new Report("Monthly Sales", "Sales increased by 20%");
        System.out.println(report.generate());
    }
}
